src/Libraries/Microsoft.Extensions.AI/ChatCompletion/CachingChatClient.cs

```diff
@@ -152,4 +237,27 @@ public override async IAsyncEnumerable<StreamingChatCompletionUpdate> CompleteSt
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests.</param>
     /// <returns>A <see cref="Task"/> representing the completion of the operation.</returns>
     protected abstract Task WriteCacheStreamingAsync(string key, IReadOnlyList<StreamingChatCompletionUpdate> value, CancellationToken cancellationToken);
+
+    /// <summary>Merges the properties from source into target.</summary>
+    /// <returns>
+    /// <paramref name="target"/> if <paramref name="target"/> is not <see langword="null"/> or if <paramref name="source"/> is <see langword="null"/> or empty.
+    /// Otherwise, a clone of <paramref name="source"/>.
+    /// </returns>
+    private static AdditionalPropertiesDictionary? MergeProperties(AdditionalPropertiesDictionary? target, AdditionalPropertiesDictionary? source)
+    {
+        if (source is { Count: > 0 })
+        {
+            if (target is null)
+            {
+                return source.Clone();
+            }
+
+            foreach (var entry in source)
+            {
+                target[entry.Key] = entry.Value;
```

Comment:
I'm not sure we should do this, but if we do, shouldn't the precedence go the other way around (as in, we have a first-chunk wins rule rather than last-chunk wins)? That would be consistent with the logic for other non-metadata properties, e.g., `coalesced.CreatedAt ??= next.CreatedAt`.